### PR TITLE
Fix command evaluation for inspec shell -c

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
Fixed #942

**Comment by ksubrama** [comment]
@stevendanna @chris-rock @arlimus @vjeffrey Yay tests pass.  I think we have a good set of functional tests.  Clearly there can be more.  I kinda want to get these in before I spend more time over testing this.

File: lib/inspec/cli.rb [hunk 1]
```diff
@@ -162,22 +162,23 @@ def shell_func
     diagnose
     o = opts.dup
 
-    log_device = opts['format'] == 'json' ? nil : STDOUT
+    json_output = ['json', 'json-min'].include?(opts['format'])
```
Comment (arlimus) [inline lib/inspec/cli.rb:165]: Great catch!!

**Comment by arlimus** [comment]
These look great, especially the large selection of additional tests. There can always be more, but the tricky part is to select the right amount of tests and I like the selection you've made.

The only slight thing we should keep in mind is this:
https://github.com/chef/inspec/pull/943/files#diff-ca3dd31f0db2b13fca008d2b3a0000b8L138
Since this is a publicly available function that we are removing, it may cause issues in other projects. I do however like that it is being removed, i.e.:

:+1: 

Thank you Kartik, this is a great improvement :smile:

**Pull request merged**